src/benchmarklib/tpcds/tpcds_table_generator.cpp

```diff
@@ -1238,117 +1239,653 @@ AbstractTableGenerator::IndexesByTable TPCDSTableGenerator::_indexes_by_table()
+                                                           {date_dim_table->column_id_by_name("d_date_sk")},
+                                                           date_dim_table});
+
+  // web_page - 1 PK, 3 FKs.
+  web_page_table->add_soft_constraint(
+      TableKeyConstraint{{web_page_table->column_id_by_name("wp_web_page_sk")}, KeyConstraintType::PRIMARY_KEY});
+
+  web_page_table->add_soft_constraint(ForeignKeyConstraint{{web_page_table->column_id_by_name("wp_creation_date_sk")},
+                                                           web_page_table,
+                                                           {date_dim_table->column_id_by_name("d_date_sk")},
+                                                           date_dim_table});
+  web_page_table->add_soft_constraint(ForeignKeyConstraint{{web_page_table->column_id_by_name("wp_access_date_sk")},
+                                                           web_page_table,
+                                                           {date_dim_table->column_id_by_name("d_date_sk")},
+                                                           date_dim_table});
+  web_page_table->add_soft_constraint(ForeignKeyConstraint{{web_page_table->column_id_by_name("wp_customer_sk")},
+                                                           web_page_table,
+                                                           {customer_table->column_id_by_name("c_customer_sk")},
+                                                           customer_table});
+
+  // warehouse - 1 PK.
+  warehouse_table->add_soft_constraint(
+      TableKeyConstraint{{warehouse_table->column_id_by_name("w_warehouse_sk")}, KeyConstraintType::PRIMARY_KEY});
+
+  // customer - 1 PK, 6 FKs.
+  customer_table->add_soft_constraint(
+      TableKeyConstraint{{customer_table->column_id_by_name("c_customer_sk")}, KeyConstraintType::PRIMARY_KEY});
+  customer_table->add_soft_constraint(
+      ForeignKeyConstraint{{customer_table->column_id_by_name("c_current_cdemo_sk")},
+                           customer_table,
+                           {customer_demographics_table->column_id_by_name("cd_demo_sk")},
+                           customer_demographics_table});
+  customer_table->add_soft_constraint(
+      ForeignKeyConstraint{{customer_table->column_id_by_name("c_current_hdemo_sk")},
+                           customer_table,
+                           {household_demographics_table->column_id_by_name("hd_demo_sk")},
+                           household_demographics_table});
+  customer_table->add_soft_constraint(ForeignKeyConstraint{{customer_table->column_id_by_name("c_current_addr_sk")},
+                                                           customer_table,
+                                                           {customer_address_table->column_id_by_name("ca_address_sk")},
+                                                           date_dim_table});
```

Comment:
Another one.

Reply:
Given these errors are super hard to spot (at least for me), I am wondering if we could use a little helper method.

`fk_pair(table, column_name) -> pair(column_id, table)` ?

Reply:
Or overleaded constructor. Whatever makes sense.

Reply:
added utility functios so this specific piece now looks like:
```c++
// customer - 1 PK, 6 FKs.
primary_key_constraint(customer_table, {"c_customer_sk"});
foreign_key_constraint(customer_table, {"c_current_cdemo_sk"}, customer_demographics_table, {"cd_demo_sk"});
foreign_key_constraint(customer_table, {"c_current_hdemo_sk"}, household_demographics_table, {"hd_demo_sk"});
foreign_key_constraint(customer_table, {"c_current_addr_sk"}, customer_address_table, {"ca_address_sk"});
foreign_key_constraint(customer_table, {"c_first_shipto_date_sk"}, date_dim_table, {"d_date_sk"});
foreign_key_constraint(customer_table, {"c_first_sales_date_sk"}, date_dim_table, {"d_date_sk"});
foreign_key_constraint(customer_table, {"c_last_review_date"}, date_dim_table, {"d_date_sk"});
```